sdk/src/jmh/java/io/opentelemetry/sdk/metrics/MetricsTestOperationBuilder.java

```diff
@@ -0,0 +1,173 @@
+                  boundMetric =
+                      meter
+                          .doubleCounterBuilder("bound_double_counter")
+                          .build()
+                          .bind("KEY", "VALUE");
+                }
+              };
+          op.initialize(meter);
+          return op;
+        }
+      }),
+  DoubleMeasureRecord(
+      new OperationBuilder<DoubleMeasure, DoubleMeasure.BoundDoubleMeasure>() {
+        @Override
+        public Operation<DoubleMeasure, DoubleMeasure.BoundDoubleMeasure> build(Meter meter) {
+          Operation<DoubleMeasure, DoubleMeasure.BoundDoubleMeasure> op =
+              new Operation<DoubleMeasure, DoubleMeasure.BoundDoubleMeasure>() {
+                @Override
+                void perform(String... args) {
+                  metric.record(5.0d, args);
+                }
+
+                @Override
+                void performBound() {
+                  boundMetric.record(5.0d);
+                }
+
+                @Override
+                protected void initialize(Meter meter) {
+                  metric = meter.doubleMeasureBuilder("double_measure").build();
+                  boundMetric =
+                      meter
+                          .doubleMeasureBuilder("bound_double_measure")
+                          .build()
+                          .bind("KEY", "VALUE");
+                }
+              };
+          op.initialize(meter);
+          return op;
+        }
+      }),
+  LongMeasureRecord(
+      new OperationBuilder<LongMeasure, LongMeasure.BoundLongMeasure>() {
+        @Override
+        public Operation<LongMeasure, LongMeasure.BoundLongMeasure> build(Meter meter) {
+          Operation<LongMeasure, LongMeasure.BoundLongMeasure> op =
+              new Operation<LongMeasure, LongMeasure.BoundLongMeasure>() {
+                @Override
+                void perform(String... args) {
+                  metric.record(5L, args);
+                }
+
+                @Override
+                void performBound() {
+                  boundMetric.record(5L);
+                }
+
+                @Override
+                protected void initialize(Meter meter) {
+                  metric = meter.longMeasureBuilder("long_measure").build();
+                  boundMetric =
+                      meter.longMeasureBuilder("bound_long_measure").build().bind("KEY", "VALUE");
+                }
+              };
+          op.initialize(meter);
+          return op;
+        }
+      });
+
+  private final OperationBuilder<?, ?> builder;
+
+  MetricsTestOperationBuilder(final OperationBuilder<?, ?> builder) {
+    this.builder = builder;
+  }
+
+  public Operation<?, ?> build(Meter meter) {
+    return this.builder.build(meter);
+  }
+
+  @Immutable
+  private interface OperationBuilder<T, U> {
+    Operation<T, U> build(Meter meter);
+  }
+
+  abstract static class Operation<T, U> {
+    T metric;
+    U boundMetric;
```

Comment:
Don't see a clear advantage to use generics here. Can we get rid of them?

Reply:
This is what's allowing us to use the metrics without an explicit cast in the Operation instances like `metric.add(5L, args)`. Otherwise both metric and boundMetric are going to need to be bare objects with casts everywhere we refer to them (which is not often. I think I prefer the generic way, but if you prefer the raw `Object` because the generics are distracting, I'm sympathetic to that argument.

Reply:
metric and metricBound can be members in the sub-class you don't need them here because you don't need access to them when calling perform and performBound.

Reply:
Of course, thank you. Tried this too many ways before and was overlooking the obvious here.